assets/js/atomic/components/product-list/sale-badge/index.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+
+const ProductListSaleBadge = ( { className, product, align } ) => {
+	const alignClass =
+		typeof align !== 'undefined'
```

Comment:
Actually we could take the chance and check that `align` is a string here:
```suggestion
		typeof align === 'string'
```